src/java.base/share/classes/sun/util/locale/LanguageTag.java

```diff
@@ -414,6 +409,48 @@ private boolean parsePrivateuse(StringTokenIterator itr, ParseStatus sts) {
         return found;
     }
 
+    public static String caseFoldTag(String tag) {
+        // Illegal tags
+        if (!parse(tag, null).wellFormed) {
+            throw new IllformedLocaleException("Ill formed tag");
```

Comment:
Good point, I have updated it so that it provides the sts error message with the exception.

For example,
`Ill formed tag: Invalid subtag: xabadadoo`